### PR TITLE
Use allThemes query from CMS to generate navigation menu

### PR DESCRIPTION
**Pull request by benweatherman**
Removed static JSON files and changed a few field names, but otherwise easy peasy. We'll only show topics if they have some published services.

jiffy

Fixes cityofaustin/techstack#105

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,20 +24,20 @@
     "react-load-script": "^0.0.6",
     "react-router": "^4.2.0",
     "react-router-dom": "^4.2.2",
-    "react-static": "^5.1.12",
+    "react-static": "^5.6.8",
```
Comment (ifsimicoded) [inline package.json]: not a big deal, just curious why the upgrade?

Comment (benweatherman) [reply package.json]: Buried in my commit message https://github.com/cityofaustin/janis/pull/210/commits/f4c147da7ede208e803036e266faf76396de5dc4

> This solves a build problem I was running into, though I'm not entirely sure why. It seems like it's a `TileGroup` or `TileGroupSet` issue, but I'm not ready to track it down.
> 
> The old version would fail the build but the new version shows the following warning
> 
```
Check the top-level render call using <div>. See https://fb.me/react-warning-keys for more information.
    in TileGroup
    in TileGroupSet
    in div
    in div
    in Theme
...
```

File: src/js/page_sections/Menu/Menu.js [hunk 2]
```diff
@@ -1,26 +1,15 @@
 import React, { Component } from 'react';
 import { FormattedMessage, defineMessages, injectIntl } from 'react-intl';
+import { withRouteData } from 'react-static';
```
Comment (ifsimicoded) [inline src/js/page_sections/Menu/Menu.js]: we can kill line 134-137 too since we don't need lang context.

File: src/__tmpdata/navigation.json [hunk 3]
```diff
@@ -1,300 +0,0 @@
-{
```
Comment (mateoclarke) [inline src/__tmpdata/navigation.json]: 💣 💥

File: src/js/page_sections/Menu/Submenu.js [hunk 4]
```diff
@@ -52,10 +52,10 @@ const ThemeSubmenuItem = ({theme, handleToggleAllMenus}) => (
     onClick={handleToggleAllMenus}
   >
     <I18nNavLink
-      to={`/theme/${theme.slug}`}
+      to={`/themes/${theme.slug}`}
```
Comment (mateoclarke) [inline src/js/page_sections/Menu/Submenu.js]: 👍

File: static.config.js [hunk 5]
```diff
@@ -21,16 +21,19 @@ const makeAllPages = async (langCode) => {
...
   console.log(`- Building routes for ${path}...`);
 
+  const client = createGraphQLClientsByLang(langCode);
+  const { allThemes } = await client.request(allThemesQuery);
+
   const data = {
     path: path,
     component: 'src/js/pages/Home',
-    children: await makeChildPages(langCode),
+    children: await makeChildPages(client, allThemes),
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply static.config.js]: Simi is right... always. I think our last convo about this may have led you down this path but this feels like the perfect use case for `allSiteData`.

[the remark this answers is not in the document]
Comment (benweatherman) [reply static.config.js]: I looked at `allSiteData` initially, but had trouble figuring out how we'd make that work with setting the language code. My thought was that we'd have to generate all language data and store it with language code as the key (see below). Then we'd have to set `languageCode` as a property and have each component get the data by that code. That seemed like pushing too much logic into components and still required us to send `languageCode` to each component, but maybe there's a better way to do it.

```
allSiteData: {
  en: {
    allThemes
  },
  es: {
    allThemesES
  },
  ...
}
```

[the remark this answers is not in the document]
Comment (ifsimicoded) [reply static.config.js]: I'm not sure I understand. Would Menu.js be the only component that needed to know language code? No less, we can inject language code as needed to components via wrapping the component in react-intl's HOC method injectIntl -- so like
```
const Submenu = ({..., intl}) => {
    let appLanguageCode = intl.locale; //<--this is the language code
    ...
}
export default injectIntl(Submenu);
```

[the remark this answers is not in the document]
Comment (benweatherman) [reply static.config.js]: Looks like they _just_ changed their docs. Curiously, `getSiteData` doesn't show up anymore :|

Good point, only Menu.js would need to use `allSiteData`. It would then need to use that language code to get the `allSiteData` that corresponds with the language it's trying to render.

```
const Menu = ({..., intl}) => {
    let appLanguageCode = intl.locale; //<--this is the language code
    let allThemes = navigationData[appLanguageCode]; // navigationData is passed in via allSiteData
}
export default injectIntl(Menu);
```

Where `getSiteData` would need to fetch all data for all languages (not just the language we're currently building)

```
getSiteData() = async () => ({
  en: createGraphQLClientsByLang('en').request(allThemesQuery),
  es: createGraphQLClientsByLang('es').request(allThemesQuery),
  vi: createGraphQLClientsByLang('vi').request(allThemesQuery),
  ar: createGraphQLClientsByLang('ar').request(allThemesQuery),
})
```

### TL;DR
Both solutions seem a little weird to me. Let me try out the `allSiteData` version and see how it nets out.

[the remark this answers is not in the document]
Comment (ifsimicoded) [reply static.config.js]: TRUTH. Thanks for the explanation and agreed. It's just going to get more complicated. And we haven't even added Esperanto.

[the remark this answers is not in the document]
Comment (benweatherman) [reply static.config.js]: Another check in the "Simi is always right" column https://github.com/cityofaustin/janis/pull/211
